src/openrct2/interface/Viewport.cpp

```diff
@@ -719,80 +713,74 @@ viewport_focus viewport_update_smart_guest_follow(rct_window* window, Peep* peep
         return focus;
     }
-    else
+
+    bool overallFocus = true;
+    if (peep->State == PeepState::OnRide || peep->State == PeepState::EnteringRide
+        || (peep->State == PeepState::LeavingRide && peep->x == LOCATION_NULL))
     {
-        bool overallFocus = true;
-        if (peep->State == PeepState::OnRide || peep->State == PeepState::EnteringRide
-            || (peep->State == PeepState::LeavingRide && peep->x == LOCATION_NULL))
+        auto ride = get_ride(peep->CurrentRide);
+        if (ride != nullptr && (ride->lifecycle_flags & RIDE_LIFECYCLE_ON_TRACK))
         {
-            auto ride = get_ride(peep->CurrentRide);
-            if (ride != nullptr && (ride->lifecycle_flags & RIDE_LIFECYCLE_ON_TRACK))
+            auto train = GetEntity<Vehicle>(ride->vehicles[peep->CurrentTrain]);
+            if (train != nullptr)
             {
-                auto train = GetEntity<Vehicle>(ride->vehicles[peep->CurrentTrain]);
-                if (train != nullptr)
+                const auto car = train->GetCar(peep->CurrentCar);
+                if (car != nullptr)
                 {
-                    const auto car = train->GetCar(peep->CurrentCar);
-                    if (car != nullptr)
-                    {
-                        focus.sprite.sprite_id = car->sprite_index;
-                        overallFocus = false;
-                    }
+                    focus.sprite.sprite_id = car->sprite_index;
+                    overallFocus = false;
                 }
             }
         }
-        if (peep->x == LOCATION_NULL && overallFocus)
-        {
-            auto ride = get_ride(peep->CurrentRide);
-            if (ride != nullptr)
-            {
-                auto xy = ride->overall_view.ToTileCentre();
-                focus.type = VIEWPORT_FOCUS_TYPE_COORDINATE;
-                focus.coordinate.x = xy.x;
-                focus.coordinate.y = xy.y;
-                focus.coordinate.z = tile_element_height(xy) + (4 * COORDS_Z_STEP);
-                focus.sprite.type |= VIEWPORT_FOCUS_TYPE_COORDINATE;
-            }
-        }
-        else
+    }
+
+    if (peep->x == LOCATION_NULL && overallFocus)
+    {
+        auto ride = get_ride(peep->CurrentRide);
+        if (ride != nullptr)
         {
-            focus.sprite.type |= VIEWPORT_FOCUS_TYPE_SPRITE | VIEWPORT_FOCUS_TYPE_COORDINATE;
-            focus.sprite.pad_486 &= 0xFFFF;
+            auto xy = ride->overall_view.ToTileCentre();
+            focus.type = VIEWPORT_FOCUS_TYPE_COORDINATE;
+            focus.coordinate.x = xy.x;
+            focus.coordinate.y = xy.y;
+            focus.coordinate.z = tile_element_height(xy) + (4 * COORDS_Z_STEP);
+            focus.sprite.type |= VIEWPORT_FOCUS_TYPE_COORDINATE;
         }
-        focus.coordinate.rotation = get_current_rotation();
     }
+    else
+    {
+        focus.sprite.type |= VIEWPORT_FOCUS_TYPE_SPRITE | VIEWPORT_FOCUS_TYPE_COORDINATE;
+        focus.sprite.pad_486 &= 0xFFFF;
```

Comment:
Seems to be unused

Reply:
It's very very easy to mess this up the union used for this has a ton of gotchas.

Reply:
Hmm, I've noticed no differences with this line removed, also haven't seen any change to `focus.location` which shares its memory.